ci/check_gucs_are_alphabetically_sorted.sh

```diff
@@ -0,0 +1,3 @@
+set -euo pipefail
```

Comment:
I think you should include these two lines:

https://github.com/citusdata/citus/blob/master/ci/banned.h.sh#L14-L15

So that when there is a failure, the job shows the README you have added for this job.

Also as the first line you can add the following so that this bash knows what to use:
```
#! /bin/bash
```

Reply:
```suggestion
set -eu
```

There are no pipes in this script so it does not really provide any value to have this shell option. 

No harm in having this extra option though. Feel free to keep it.

Reply:
This is like a template that we have in our scripts so that there is a standard and later if we modify the script to add a pipe we will already have the check. 

If remove this check now, I think it is likely that we will forget to add the pipefail option when adding a new command which uses a pipe.